OSPP Proposal: PIPL-Compliant Cloud-Edge Collaborative Privacy-Preserving Prompt Processing Framework on KubeEdge-Ianvs

Hanyuan9766 opened the pull request:
…dge=Llama-3-8B-Instruct and MIA eval (#203)

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://github.com/kubeedge/ianvs/blob/main/CONTRIBUTING.md
2. Ensure you have added or ran the appropriate tests for your PR

-->

**What type of PR is this?**

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature
/kind test
/kind design

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind failing-test
-->


**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

MooreZheng commented on docs/proposals/scenarios/privacy-preserving-llm-collaboration/optimizing-privacy-performance-equilibrium-cloud-edge-llm-systems_en_PR.md:
```diff
@@ -0,0 +1,215 @@
+
+#### **System Component Architecture**
+
+![System Component Architecture](<./images/System Component Architecture.png>)
```
DO NOT change the name of key components of KubeEdge Ianvs. 
It should be Test environment manager, test case controller and story manager. Any change should be considered in it. See example from others if you also want to add a edge-cloud one https://github.com/kubeedge/ianvs/blob/main/docs/proposals/scenarios/government_affairs/government_affairs.md

MooreZheng commented on docs/proposals/scenarios/privacy-preserving-llm-collaboration/optimizing-privacy-performance-equilibrium-cloud-edge-llm-systems_en_PR.md:
```diff
@@ -0,0 +1,215 @@
+  * Packet inspection verifies that the cross-border payload strictly adheres to the "anonymized vector + minimal tags" structure.  
+  * Unit tests cover scenarios ensuring no raw text or PII is transmitted cross-border.  
+  * Edge/cloud audit logs contain all critical parameters and verification hashes.
+
```
Dataset is also the key contribution. Need decription of the new dataset showing what is going to be included in it. For your reference https://www.kaggle.com/datasets/kubeedgeianvs/pcb-aoi

Commit:
docs: add PIPL-compliant privacy-preserving LLM collaboration framework

Add comprehensive framework for privacy-preserving LLM collaboration between cloud and edge:

- Two architecture diagrams (End-to-End Workflow, System Component Architecture)
- English PR document with technical specifications and implementation details
- Chinese PR document with equivalent content for broader accessibility
- PIPL compliance framework with edge-cloud collaborative inference
- GPT-4o-mini cloud model integration with Llama-3-8B-Instruct edge processing

This addresses PR #266 for OSPP privacy-preserving LLM collaboration scenario.

Signed-off-by: Hanyuan9766 <zhy2327@163.com>

Commit:
revert: remove PR #266 implementation files, keep only documentation

- Remove cloud-edge-collaborative-inference-for-llm implementation directory
- Keep only documentation files as requested:
  - docs/proposals/scenarios/privacy-preserving-llm-collaboration/images/End-to-End Workflow.png
  - docs/proposals/scenarios/privacy-preserving-llm-collaboration/images/System Component Architecture.png
  - docs/proposals/scenarios/privacy-preserving-llm-collaboration/optimizing-privacy-performance-equilibrium-cloud-edge-llm-systems_en_PR.md
  - docs/proposals/scenarios/privacy-preserving-llm-collaboration/optimizing-privacy-performance-equilibrium-cloud-edge-llm-systems_zh_PR.md

This reverts the implementation part of PR #266 while preserving the documentation.

Signed-off-by: Hanyuan9766 <zhy2327@163.com>

Commit:
restore: download PR #266 files back to local

- Restore OSPP-implemention-Proposal-PIPL-Compliant-Cloud-Edge-Collaborative-Privacy-Preserving-Prompt-Processing-Framework-203 implementation files
- Keep existing documentation files intact
- All files from PR #266 are now available locally

This restores the complete implementation from PR #266 while maintaining
the documentation files that were preserved during the rollback.

Signed-off-by: Hanyuan9766 <zhy2327@163.com>

Commit:
cleanup: remove redundant documentation files from PR #278

- Remove docs/proposals/scenarios/privacy-preserving-llm-collaboration/ directory
- Keep only the core implementation files for PIPL privacy framework
- This ensures the PR contains only the essential implementation code

Signed-off-by: Hanyuan9766 <zhy2327@163.com>

Hanyuan9766 committed:
chore(examples): add COMMIT_PROOF_PR278 for authorship verification (no issue refs)

Signed-off-by: Hanyuan9766 <zhylife2327@163.com>

Hanyuan9766 committed:
feat: comprehensive update to PR #278 PIPL framework implementation

- Enhanced benchmarkingjob.yaml with detailed description
- Updated requirements.txt with IANVS integration and additional privacy metrics
- Improved privacy_preserving_llm.py with comprehensive documentation and features
- Added zero raw text cross-border transmission capabilities
- Integrated Membership Inference Attack (MIA) evaluation framework
- Enhanced PIPL compliance verification and audit logging

This update significantly improves the PR #278 implementation with better
documentation, enhanced features, and comprehensive privacy protection.

Addresses issue 203

Signed-off-by: 寮犵€氬厓 <zhylife2327@163.com>

Hanyuan9766 committed:
feat: force update PR #278 with complete PIPL framework

- Force update to ensure PR #278 reflects latest changes
- Comprehensive PIPL-compliant cloud-edge collaborative framework
- Enhanced documentation and implementation
- Zero raw text cross-border transmission
- Membership Inference Attack evaluation framework

This commit ensures PR #278 is properly updated with all latest changes.

Addresses issue 203

Signed-off-by: 寮犵€氬厓 <zhylife2327@163.com>

Hanyuan9766 committed:
restore: keep existing example 'cloud-edge-collaborative-inference-for-LLM' per review

- Restore directory from upstream/main
- Ensure PR #278 does not delete existing example

Signed-off-by: 寮犵€氬厓 <zhylife2327@163.com>
